docker/docker.ts

```diff
@@ -315,28 +343,25 @@ async function dockerBuild(
         buildArgs.push(...[ "--target", target ]);
     }
 
-    const buildResult = await runCLICommand("docker", buildArgs, logResource, /*reportStdout*/ true);
-    if (buildResult.code) {
-        throw new ResourceError(`Docker build of image '${imageName}' failed with exit code: ${buildResult.code}`, logResource);
-    }
+    await runCommandThatMustSucceed("docker", buildArgs, logResource);
 }
 
 async function loginToRegistry(registry: Registry, logResource: pulumi.Resource): Promise<void> {
     const { registry: registryName, username, password } = registry;
 
     const dockerPasswordStdin = await useDockerPasswordStdin(logResource);
 
-    let loginResult: CommandResult;
+    // pass 'reportFullCommandLine: false' here so that if we fail to login we don't emit the
+    // username/password in our logs.  Instead, we'll just say "'docker login' failed with code ..."
     if (!dockerPasswordStdin) {
-        loginResult = await runCLICommand(
-            "docker", ["login", "-u", username, "-p", password, registryName], logResource, /*reportStdout*/ true);
-    } else {
-        loginResult = await runCLICommand(
-            "docker", ["login", "-u", username, "--password-stdin", registryName],
-            logResource, /*reportStdout*/ true, password);
+        await runCommandThatMustSucceed(
+            "docker", ["login", "-u", username, "-p", password, registryName],
+            logResource, /*reportFullCommandLine*/ false);
     }
-    if (loginResult.code) {
-        throw new ResourceError(`Failed to login to Docker registry ${registryName}`, logResource);
+    else {
+        await runCommandThatMustSucceed(
+            "docker", ["login", "-u", username, "--password-stdin", registryName],
+            logResource, /*reportFullCommandLine*/ false, password);
```

Comment:
for hte login commands we pass `reportFullCommandLine:false` so that the messages we get are things like `Executing 'docker login'`, and we don't print out things like the actual username/password.